test/PresentationExchange.spec.ts

```diff
@@ -123,9 +169,68 @@ function getVCs(): IVerifiableCredential[] {
 }
 
 describe('presentation exchange manager tests', () => {
-  it("validatePresentationAgainstDefinition: should throw error if provided VP doesn't match the PD", async function () {
-    const payload: AuthenticationRequestPayload = await getPayload();
-    const pd: PresentationDefinitionWithLocation[] = PresentationExchange.findValidPresentationDefinitions(payload);
+  it("validatePresentationAgainstDefinition: should throw error if provided VP doesn't match the PD val", async function () {
+    const payload: AuthenticationRequestPayload = await getPayload_PD_Val();
+    const pd: PresentationDefinitionWithLocation[] = await PresentationExchange.findValidPresentationDefinitions(payload);
+    const vcs = getVCs();
+    vcs[0].issuer = { id: 'did:example:totallyDifferentIssuer' };
+    const verifiedJwt: SIOP.VerifiedAuthenticationRequestWithJWT = {
+      didResolutionResult: undefined,
+      issuer: '',
+      jwt: '',
+      signer: undefined,
+      payload: payload,
+      presentationDefinitions: pd,
+      verifyOpts: null,
+    };
+    try {
+      await PresentationExchange.validatePresentationAgainstDefinition(verifiedJwt.presentationDefinitions[0].definition, {
+        '@context': ['https://www.w3.org/2018/credentials/v1'],
+        type: ['VerifiablePresentation'],
+        verifiableCredential: vcs,
+      });
+    } catch (e) {
+      expect(e.message).toContain(SIOPErrors.COULD_NOT_FIND_VCS_MATCHING_PD);
```

Comment:
You can also put the expect on the validate function and then use .rejects instead of catching the Error and then doin an expect btw.

Reply:
changed